lib/components/map/itinerary-summary-overlay.tsx

```diff
@@ -0,0 +1,223 @@
+import { connect } from 'react-redux'
+import { Feature, lineString, LineString, Position } from '@turf/helpers'
+import { Itinerary, Location } from '@opentripplanner/types'
+import { Marker } from 'react-map-gl'
+import centroid from '@turf/centroid'
+import distance from '@turf/distance'
+import polyline from '@mapbox/polyline'
+import React, { useContext, useState } from 'react'
+import styled from 'styled-components'
+
+import * as narriativeActions from '../../actions/narrative'
+import { AppReduxState } from '../../util/state-types'
+import { boxShadowCss } from '../form/batch-styled'
+import { ComponentContext } from '../../util/contexts'
+import { doMergeItineraries } from '../narrative/narrative-itineraries'
+import {
+  getActiveItinerary,
+  getActiveSearch,
+  getVisibleItineraryIndex
+} from '../../util/state'
+import MetroItineraryRoutes from '../narrative/metro/metro-itinerary-routes'
+
+type ItinWithGeometry = Itinerary & { allLegGeometry: Feature<LineString> }
+
+type Props = {
+  from: Location
+  itins: Itinerary[]
+  setActiveItinerary: ({ index }: { index: number | null }) => void
+  setVisibleItinerary: ({ index }: { index: number | null }) => void
+  to: Location
+  visible?: boolean
+  visibleItinerary?: number
+}
+
+const Card = styled.div`
+  ${boxShadowCss}
+
+  background: #fffffffa;
+  border-radius: 5px;
+  padding: 5px;
+  align-items: center;
+  display: flex;
+  flex-wrap: wrap;
+
+  
+  span {
+    span {
+      span {
+        max-height: 28px;
+        min-height: 20px;
+      }
+    }
+  }
+  div {
+    margin-top: -0px!important;
+  }
+  .route-block-wrapper span {
+    padding: 0px;
+  }
+  * {
+    height: 27px;
+  }
+}
+`
+
+function addItinLineString(itin: Itinerary): ItinWithGeometry {
+  return {
+    ...itin,
+    allLegGeometry: lineString(
+      itin.legs.flatMap((leg) => polyline.decode(leg.legGeometry.points))
+    )
+  }
+}
+
+type ItinUniquePoint = {
+  itin: Itinerary
+  uniquePoint: Position
+}
+
+function getUniquePoint(
```

Comment:
Is there a way we could add padding to the start and end of the line, so that the badge isn't ever showing up on top of the origin or destination?

![image](https://github.com/opentripplanner/otp-react-redux/assets/115499534/47932db4-0cf5-4e46-9fd7-4217547e07aa)

Reply:
I tried to add another min to the minmax but it screwed up in too many cases. This doesn't happen too often and it's fewer cases than the other one failed in